### ISSUE TITLE
Param/Value passing

### ISSUE THREAD

**Issue by wandell** [isetbio/isetbio#54]
What are the new standards for setting the Param/Value arguments and argument parsing?

**Comment by DavidBrainard** [comment]
I wrote up an exampleSet function that shows the use of Matlab’s input parser and that implements one possible scheme for standards (see its comments).  This is in isettools/codedev/exampleSet.m.  An exampleGet would be the same, except that there would be no required value argument, just the field followed by key/value optional parameters.

The examples shows how one could implement default units using the parser object.  Anyway, I think the scheme in the example would be easily folded into the outsegment set/get methods James is implementing and also applied to create and plot methods.

It is possible that it would make sense for the code that sets up the parser for the key/value pairs to be common between set and get methods and just called for each, which would guarantee that sets and gets understood key/values in the same way.

In any case, why don’t interested parties have a look at the example and we can continue the dialog from there.

As it happens, I inserted the example into both the master and outersegment branches, not particularly because I initially intended to.

> On Jul 18, 2015, at 8:24 AM, Brian Wandell <notifications@github.com> wrote:
> 
> What are the new standards for setting the Param/Value arguments and argument parsing?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/isetbio/isetbio/issues/54>.
> 

David H. Brainard
Professor of Psychology
Director, Penn Vision Research Center
Director, Institute for Research in Cognitive Science
University of Pennsylvania
3401 Walnut Street, Suite 400A (mail address)
3401 Walnut Street, Room 315C (office location)
Philadelphia, PA, 19104
Phone: 215-573-7579
Email: brainard@psych.upenn.edu
Web: http://color.psych.upenn.edu/brainard
